Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyPostedInvoiceExport.Codeunit.al

```diff
@@ -0,0 +1,423 @@
+            if ShpfyVariant.IsEmpty() then
+                exit(false);
+        end;
+
+        exit(true);
+    end;
+
+    local procedure ProductVariantExists(UnitOfMeasure: Code[10]; SalesInvoiceLine: Record "Sales Invoice Line"): Boolean
+    var
+        ShpfyVariant: Record "Shpfy Variant";
+    begin
+        ShpfyVariant.SetRange("Item No.", SalesInvoiceLine."No.");
+        ShpfyVariant.SetRange("Variant Code", SalesInvoiceLine."Variant Code");
+        if ShpfyVariant.FindSet() then
+            repeat
+                case ShpfyVariant."UoM Option Id" of
+                    1:
+                        if ShpfyVariant."Option 1 Value" = UnitOfMeasure then
+                            exit(true);
+                    2:
+                        if ShpfyVariant."Option 2 Value" = UnitOfMeasure then
+                            exit(true);
+                    3:
+                        if ShpfyVariant."Option 3 Value" = UnitOfMeasure then
+                            exit(true);
+                end;
+            until ShpfyVariant.Next() = 0;
+    end;
+
+    local procedure MapPostedSalesInvoiceData(
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary;
+        var TempShpfyOrderLine: Record "Shpfy Order Line" temporary;
+        var ShpfyOrderTaxLines: Dictionary of [Text, Decimal]
+    )
+    var
+        InvoiceLine: Record "Sales Invoice Line";
+    begin
+        MapSalesInvoiceHeader(SalesInvoiceHeader, TempShpfyOrderHeader);
+
+        InvoiceLine.SetRange("Document No.", SalesInvoiceHeader."No.");
+        if InvoiceLine.FindSet() then
+            repeat
+                MapSalesInvoiceLine(InvoiceLine, TempShpfyOrderHeader, TempShpfyOrderLine, ShpfyOrderTaxLines);
+            until InvoiceLine.Next() = 0;
+    end;
+
+    local procedure MapSalesInvoiceHeader(
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary
+    )
+    begin
+        TempShpfyOrderHeader.Init();
+        TempShpfyOrderHeader."Sales Invoice No." := SalesInvoiceHeader."No.";
+        TempShpfyOrderHeader."Sales Order No." := SalesInvoiceHeader."Order No.";
+        TempShpfyOrderHeader."Created At" := SalesInvoiceHeader.SystemCreatedAt;
+        TempShpfyOrderHeader.Confirmed := true;
+        TempShpfyOrderHeader."Updated At" := SalesInvoiceHeader.SystemModifiedAt;
+        TempShpfyOrderHeader."Currency Code" := MapCurrencyCode(SalesInvoiceHeader);
+        TempShpfyOrderHeader."Document Date" := SalesInvoiceHeader."Document Date";
+        SalesInvoiceHeader.CalcFields(Amount, "Amount Including VAT", "Invoice Discount Amount");
+        TempShpfyOrderHeader."VAT Amount" := SalesInvoiceHeader."Amount Including VAT" - SalesInvoiceHeader.Amount;
+
+        MapBillInformation(TempShpfyOrderHeader, SalesInvoiceHeader);
+        MapShipToInformation(TempShpfyOrderHeader, SalesInvoiceHeader);
```

Comment:
You have Ship-to and bill-to, what about sell-to?

Reply:
I was thinking this, but in the draft order create there is nowhere to map this